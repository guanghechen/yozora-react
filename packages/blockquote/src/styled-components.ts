import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
