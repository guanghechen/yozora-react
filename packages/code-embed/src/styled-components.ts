import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
