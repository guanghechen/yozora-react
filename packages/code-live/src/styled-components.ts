import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
