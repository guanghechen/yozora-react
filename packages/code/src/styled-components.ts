import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
