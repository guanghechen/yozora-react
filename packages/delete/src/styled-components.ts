import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
