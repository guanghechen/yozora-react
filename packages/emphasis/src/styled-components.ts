import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
