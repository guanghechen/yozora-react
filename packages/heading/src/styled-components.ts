import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
