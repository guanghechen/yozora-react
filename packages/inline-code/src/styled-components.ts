import 'styled-components'
import type { YozoraInlineCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    inlineCode: YozoraInlineCodeTheme
  }
}
