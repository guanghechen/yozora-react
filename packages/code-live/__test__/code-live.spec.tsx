import React from 'react'
import { act } from 'react-dom/test-utils'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import CodeLive, { CodeRendererProps } from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  const JsxRenderer = ({ value }: CodeRendererProps): React.ReactElement => {
    // eslint-disable-next-line no-new-func
    const f = new Function(value)
    const v = f()
    return <span data-type="jsx">{ v }</span>
  }

  it('forward ref', () => {
    const ref = React.createRef<HTMLDivElement>()
    const wrapper = mount(
      <CodeLive
        ref={ ref }
        lang="jsx"
        value="const a = 1 + 2; return a;"
        CodeRenderer={ JsxRenderer }
        data-value="waw"
      />
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('change and debounce', async () => {
    const code1 = 'return 3'
    const code2 = 'return 4'

    const wrapper = mount(
      <CodeLive
        lang="jsx"
        value={ code1 }
        CodeRenderer={ JsxRenderer }
      />
    )

    expect(wrapper.find('textarea').text()).toEqual(code1)
    expect(wrapper.find('[data-type="jsx"]').text()).toEqual('3')

    // change code
    wrapper.find('textarea').simulate('change', { target: { value: code2 } })
    expect(wrapper.find('textarea').text()).toEqual(code2)
    expect(wrapper.find('[data-type="jsx"]').text()).toEqual('3')

    // await debounce
    await act(async () => {
      await new Promise(resolve => setTimeout(resolve, 1000))
    })
    expect(wrapper.find('textarea').text()).toEqual(code2)
    expect(wrapper.find('[data-type="jsx"]').text()).toEqual('4')
  })

  it('snapshot', () => {
    const code = `
      const a = 1 + 2;
      return a * a
    `

    const wrapper = render(
      <CodeLive
        lang="jsx"
        value={ code }
        CodeRenderer={ JsxRenderer }
      />
    )

    expect(wrapper.find('textarea').text()).toEqual(code)
    expect(wrapper.find('[data-type="jsx"]').text()).toEqual('9')
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        codeLive: {
          margin: '1rem 0',
          editorBackground: 'pink',
          editorCaretColor: 'white',
          editorFontSize: '18px',
        },
        codeEmbed: {
          errorBackground: 'red',
        }
      }
    }

    const code = `
      const a = 1 + 2;
      return a * a
    `

    const wrapper = render(
      <ThemeProvider theme={ theme }>
        <CodeLive
          lang="jsx"
          value={ code }
          CodeRenderer={ JsxRenderer }
        />
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
