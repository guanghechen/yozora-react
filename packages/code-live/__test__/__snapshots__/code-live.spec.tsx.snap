// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c5 {
  text-indent: -2.5em;
  padding-left: 2.5em;
}

.c6 {
  display: inline-block;
  width: 2.5em;
  padding-right: 0.6em;
  color: #858585;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: right;
}

.c4:empty {
  -webkit-text-fill-color: inherit !important;
}

.c3 {
  white-space: pre;
  font-family: Consolas,"Source Code Pro",monospace,sans-serif;
}

.c3 > pre code,
.c3 > pre span {
  line-height: inherit !important;
}

.c3 > pre code {
  background: transparent !important;
  margin: 0 !important;
  padding: 0 !important;
}

.c11 {
  position: absolute;
  top: 0;
  left: 0;
  display: none;
  width: 100%;
  height: 100%;
  overflow-y: auto;
}

.c9 {
  position: relative;
  border: none;
  padding: 0;
  background: transparent;
  color: inherit;
}

.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  overflow: auto;
  padding: 0.8rem 0;
  background: #1e1e1e;
  caret-color: #aeafad;
  font-size: 1rem;
  font-family: Consolas,"Source Code Pro",monospace,sans-serif;
}

.c2 pre::selection,
.c2 textarea::selection {
  background: none;
}

.c10 {
  width: 100%;
  height: 100%;
  border: none;
  padding: 0;
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: block;
  overflow: auto;
  color: #000;
  background: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c0 .c1,
.c0 .c7 {
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  width: 50%;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4 {
    color: transparent !important;
  }

  .c4::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}

<div
  class="c0"
  value="
      const a = 1 + 2;
      return a * a
    "
>
  <div
    class="c1 c2"
  >
    <div
      class="c3"
      style="position:relative;text-align:left;box-sizing:border-box;padding:0;overflow:hidden"
    >
      <textarea
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        class="c4"
        data-gramm="false"
        spellcheck="false"
        style="margin:0;border:0;background:none;box-sizing:inherit;display:inherit;font-family:inherit;font-size:inherit;font-style:inherit;font-variant-ligatures:inherit;font-weight:inherit;letter-spacing:inherit;line-height:inherit;tab-size:2;text-indent:inherit;text-rendering:inherit;text-transform:inherit;white-space:pre-wrap;word-break:normal;overflow-wrap:break-word;position:absolute;top:0;left:0;height:100%;width:100%;resize:none;color:inherit;overflow:hidden;-moz-osx-font-smoothing:grayscale;-webkit-font-smoothing:antialiased;-webkit-text-fill-color:transparent;padding-top:10px;padding-right:10px;padding-bottom:10px;padding-left:2.5em;word-spacing:normal;word-wrap:break-word"
      >
        
      const a = 1 + 2;
      return a * a
    
      </textarea>
      <pre
        aria-hidden="true"
        style="margin:0;border:0;background:none;box-sizing:inherit;display:inherit;font-family:inherit;font-size:inherit;font-style:inherit;font-variant-ligatures:inherit;font-weight:inherit;letter-spacing:inherit;line-height:inherit;tab-size:2;text-indent:inherit;text-rendering:inherit;text-transform:inherit;white-space:pre-wrap;word-break:normal;overflow-wrap:break-word;position:relative;pointer-events:none;padding-top:10px;padding-right:10px;padding-bottom:10px;padding-left:0;word-spacing:normal;word-wrap:break-word"
      >
        <div
          class="c5 token-line"
          style="color:#9CDCFE"
        >
          <span
            class="c6"
          >
            1
          </span>
          <span
            class="token plain"
            style="display:inline-block"
          />
        </div>
        <div
          class="c5 token-line"
          style="color:#9CDCFE"
        >
          <span
            class="c6"
          >
            2
          </span>
          <span
            class="token plain"
          >
                  
          </span>
          <span
            class="token keyword"
            style="color:rgb(86, 156, 214)"
          >
            const
          </span>
          <span
            class="token plain"
          >
             a 
          </span>
          <span
            class="token operator"
            style="color:rgb(212, 212, 212)"
          >
            =
          </span>
          <span
            class="token plain"
          >
             
          </span>
          <span
            class="token number"
            style="color:rgb(181, 206, 168)"
          >
            1
          </span>
          <span
            class="token plain"
          >
             
          </span>
          <span
            class="token operator"
            style="color:rgb(212, 212, 212)"
          >
            +
          </span>
          <span
            class="token plain"
          >
             
          </span>
          <span
            class="token number"
            style="color:rgb(181, 206, 168)"
          >
            2
          </span>
          <span
            class="token punctuation"
            style="color:rgb(212, 212, 212)"
          >
            ;
          </span>
          <span
            class="token plain"
          />
        </div>
        <div
          class="c5 token-line"
          style="color:#9CDCFE"
        >
          <span
            class="c6"
          >
            3
          </span>
          <span
            class="token plain"
          >
                  
          </span>
          <span
            class="token keyword"
            style="color:rgb(86, 156, 214)"
          >
            return
          </span>
          <span
            class="token plain"
          >
             a 
          </span>
          <span
            class="token operator"
            style="color:rgb(212, 212, 212)"
          >
            *
          </span>
          <span
            class="token plain"
          >
             a
          </span>
        </div>
        <div
          class="c5 token-line"
          style="color:#9CDCFE"
        >
          <span
            class="c6"
          >
            4
          </span>
          <span
            class="token plain"
          >
                
          </span>
        </div>
      </pre>
    </div>
  </div>
  <div
    class="c7 c8"
  >
    <div
      class="c9 c10"
    >
      <span
        data-type="jsx"
      >
        9
      </span>
      <div
        class="c11"
        hidden=""
      />
    </div>
  </div>
</div>
`;

exports[`basic rendering case snapshot with theme 1`] = `
.c5 {
  text-indent: -2.5em;
  padding-left: 2.5em;
}

.c6 {
  display: inline-block;
  width: 2.5em;
  padding-right: 0.6em;
  color: #858585;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: right;
}

.c4:empty {
  -webkit-text-fill-color: inherit !important;
}

.c3 {
  white-space: pre;
  font-family: Consolas,"Source Code Pro",monospace,sans-serif;
}

.c3 > pre code,
.c3 > pre span {
  line-height: inherit !important;
}

.c3 > pre code {
  background: transparent !important;
  margin: 0 !important;
  padding: 0 !important;
}

.c11 {
  position: absolute;
  top: 0;
  left: 0;
  display: none;
  width: 100%;
  height: 100%;
  overflow-y: auto;
}

.c9 {
  position: relative;
  border: none;
  padding: 0;
  background: transparent;
  color: inherit;
}

.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  overflow: auto;
  padding: 0.8rem 0;
  background: pink;
  caret-color: white;
  font-size: 18px;
  font-family: Consolas,"Source Code Pro",monospace,sans-serif;
}

.c2 pre::selection,
.c2 textarea::selection {
  background: none;
}

.c10 {
  width: 100%;
  height: 100%;
  border: none;
  padding: 0;
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: block;
  overflow: auto;
  color: #000;
  background: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c0 .c1,
.c0 .c7 {
  -webkit-flex: 1 1 0;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
  width: 50%;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .c4 {
    color: transparent !important;
  }

  .c4::selection {
    background-color: #accef7 !important;
    color: transparent !important;
  }
}

<div
  class="c0"
  value="
      const a = 1 + 2;
      return a * a
    "
>
  <div
    class="c1 c2"
  >
    <div
      class="c3"
      style="position:relative;text-align:left;box-sizing:border-box;padding:0;overflow:hidden"
    >
      <textarea
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        class="c4"
        data-gramm="false"
        spellcheck="false"
        style="margin:0;border:0;background:none;box-sizing:inherit;display:inherit;font-family:inherit;font-size:inherit;font-style:inherit;font-variant-ligatures:inherit;font-weight:inherit;letter-spacing:inherit;line-height:inherit;tab-size:2;text-indent:inherit;text-rendering:inherit;text-transform:inherit;white-space:pre-wrap;word-break:normal;overflow-wrap:break-word;position:absolute;top:0;left:0;height:100%;width:100%;resize:none;color:inherit;overflow:hidden;-moz-osx-font-smoothing:grayscale;-webkit-font-smoothing:antialiased;-webkit-text-fill-color:transparent;padding-top:10px;padding-right:10px;padding-bottom:10px;padding-left:2.5em;word-spacing:normal;word-wrap:break-word"
      >
        
      const a = 1 + 2;
      return a * a
    
      </textarea>
      <pre
        aria-hidden="true"
        style="margin:0;border:0;background:none;box-sizing:inherit;display:inherit;font-family:inherit;font-size:inherit;font-style:inherit;font-variant-ligatures:inherit;font-weight:inherit;letter-spacing:inherit;line-height:inherit;tab-size:2;text-indent:inherit;text-rendering:inherit;text-transform:inherit;white-space:pre-wrap;word-break:normal;overflow-wrap:break-word;position:relative;pointer-events:none;padding-top:10px;padding-right:10px;padding-bottom:10px;padding-left:0;word-spacing:normal;word-wrap:break-word"
      >
        <div
          class="c5 token-line"
          style="color:#9CDCFE"
        >
          <span
            class="c6"
          >
            1
          </span>
          <span
            class="token plain"
            style="display:inline-block"
          />
        </div>
        <div
          class="c5 token-line"
          style="color:#9CDCFE"
        >
          <span
            class="c6"
          >
            2
          </span>
          <span
            class="token plain"
          >
                  
          </span>
          <span
            class="token keyword"
            style="color:rgb(86, 156, 214)"
          >
            const
          </span>
          <span
            class="token plain"
          >
             a 
          </span>
          <span
            class="token operator"
            style="color:rgb(212, 212, 212)"
          >
            =
          </span>
          <span
            class="token plain"
          >
             
          </span>
          <span
            class="token number"
            style="color:rgb(181, 206, 168)"
          >
            1
          </span>
          <span
            class="token plain"
          >
             
          </span>
          <span
            class="token operator"
            style="color:rgb(212, 212, 212)"
          >
            +
          </span>
          <span
            class="token plain"
          >
             
          </span>
          <span
            class="token number"
            style="color:rgb(181, 206, 168)"
          >
            2
          </span>
          <span
            class="token punctuation"
            style="color:rgb(212, 212, 212)"
          >
            ;
          </span>
          <span
            class="token plain"
          />
        </div>
        <div
          class="c5 token-line"
          style="color:#9CDCFE"
        >
          <span
            class="c6"
          >
            3
          </span>
          <span
            class="token plain"
          >
                  
          </span>
          <span
            class="token keyword"
            style="color:rgb(86, 156, 214)"
          >
            return
          </span>
          <span
            class="token plain"
          >
             a 
          </span>
          <span
            class="token operator"
            style="color:rgb(212, 212, 212)"
          >
            *
          </span>
          <span
            class="token plain"
          >
             a
          </span>
        </div>
        <div
          class="c5 token-line"
          style="color:#9CDCFE"
        >
          <span
            class="c6"
          >
            4
          </span>
          <span
            class="token plain"
          >
                
          </span>
        </div>
      </pre>
    </div>
  </div>
  <div
    class="c7 c8"
  >
    <div
      class="c9 c10"
    >
      <span
        data-type="jsx"
      >
        9
      </span>
      <div
        class="c11"
        hidden=""
      />
    </div>
  </div>
</div>
`;
